tests/components/weather/test_weather.py

```diff
@@ -0,0 +1,41 @@
+"""The tests for the Weather component."""
+import unittest
+from unittest import mock
+
+from homeassistant.components import weather
+from homeassistant.util.unit_system import METRIC_SYSTEM
+from homeassistant.bootstrap import setup_component
+
+from tests.common import get_test_home_assistant
+
+
+class TestWeather(unittest.TestCase):
+    """Test the weather class."""
+
+    def setUp(self):
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+        self.hass.config.units = METRIC_SYSTEM
+        self.assertTrue(setup_component(self.hass, weather.DOMAIN, {
+            'weather': {
+                'platform': 'demo',
+            }}))
+
+    def tearDown(self):
+        """Stop down everything that was started."""
+        self.hass.stop()
+
+    def test_source_attributes(self):
```

Comment:
I would expect a simple test to make sure that all properties are written as state attributes to the state machine. So for example setup the demo platform and do something like:

```python
state = self.hass.states.get('weather.demo')
assert state is not None
data = state.attributes
assert data.get(ATTR_TEMPERATURE) == 10
# etc
```